### PR TITLE
Add extra cypress tests

### PR DESCRIPTION
**Pull request by Toxiapo**
This PR adds extra cypress tests to the HTML renderer section and also validates other tests, which should cover the following test scenarios

- [x] when you scroll to the bottom of the chapter and press "next", it knows that it's at the end of the chapter and loads the next one
- [x] when you scroll, then resize, it knows you're not yet at the end of the chapter
- [x] Internal links works on both paginated and scrolling mode
- [x] TOC navigates on paginated and scrolling mode
- [x] Height and width problem, make sure there are no extra scrollbars
- [x] All the settings under the settings button
- [x] Next / Previous button response to book state (navigates + hide/show state)

### CODE REVIEW

**Review by samanthaandrews — APPROVED** [review]
nice!

File: cypress/integration/html-reader/page-button.ts [hunk 1]
```diff
@@ -119,4 +138,42 @@ describe('PageButton visibility on useHtmlReader', () => {
...
+      'not.be.disabled'
+    );
+
+    cy.log('Small screen should disable next button');
+    cy.viewport(100, 100);
+
+    cy.findByRole('button', { name: 'Next Page' }).should('be.disabled');
+    cy.findByRole('button', { name: 'Previous Page' }).should(
+      'not.be.disabled'
+    );
```
Comment (kristojorg) [inline cypress/integration/html-reader/page-button.ts:177]: I don't think this is testing what we want. We want to test that screen resize will trigger an update on the disabled state of the prev/next buttons. In this case, they stay the same.

**Review by kristojorg — CHANGES_REQUESTED** [review]
This looks great, just the one test I think we should alter a bit!

File: cypress/integration/html-reader/page-button.ts (hunk 1 above)
Comment (Toxiapo) [reply cypress/integration/html-reader/page-button.ts:177]: Hmm okay, this test is just not correct actually, next page button should always be disabled on scrolling mode cuz it's on the last page(For the current webreader version, it's chapter navigation). This is only true for paginated mode where a resize will trigger the next/previous state, we already have a test for that.  

Butttt, I think this is different in the new HTML renderer? I can make a comment to change this after the merge if this is what we decided to do.

Comment (kristojorg) [reply cypress/integration/html-reader/page-button.ts:177]: I think the new renderer and the old will function the same. When in scrolling mode, the next/prev button should take you one page unit down the page in scrolling mode (as well as paginated mode)

Comment (Toxiapo) [reply cypress/integration/html-reader/page-button.ts:177]: I think we might have found a bug. It's chapter navigation for me on the current version.

I am going to file a ticket for this bug to fix this and monitor the bug after we merge #91

Comment (Toxiapo) [reply cypress/integration/html-reader/page-button.ts:177]: Ticket: [OE-376](https://jira.nypl.org/browse/OE-376)

**Pull request merged**